pydoctor/model.py

```diff
@@ -546,6 +547,19 @@ class Attribute(Inheritable):
 
 T = TypeVar('T')
 
+
+# Declare the types that we consider as functions (also when they are coming
+# from a C extension)
+func_types = [types.BuiltinFunctionType, types.FunctionType]
+if hasattr(types, "MethodDescriptorType"):
+    # This is Python >= 3.7 only
+    func_types.append(types.MethodDescriptorType)
+if hasattr(types, "ClassMethodDescriptorType"):
+    # This is Python >= 3.7 only
+    func_types.append(types.ClassMethodDescriptorType)
+func_types = tuple(func_types)
```

Comment:
> Static checks are complaining about the fact that `func_types` (which is used in an `isinstance()` call) is first built as a list and then later turned into a tuple. I can change it to being a tuple from the very beginning if that's a problem.

Yes, we do want all static checks to pass. A possible solution would be to define `iter_func_types()` and capture the result of that in a tuple.